src/Monitoring.GitHubVulnerabilitiesLag/Job.cs

```diff
@@ -0,0 +1,160 @@
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Design;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Autofac;
+using Autofac.Extensions.DependencyInjection;
+using Microsoft.ApplicationInsights;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.WindowsAzure.Storage;
+using NuGet.Services.Configuration;
+using NuGet.Services.Cursor;
+using NuGet.Services.KeyVault;
+using NuGet.Services.Logging;
+using NuGet.Services.Storage;
+
+namespace NuGet.Jobs.Monitoring.GitHubVulnerabilitiesLag
+{
+    public class Job : JobBase
+    {
+        private const string ConfigurationArgument = "Configuration";
+        private const string MonitorConfigurationSectionName = "MonitorConfiguration";
+
+        private static readonly TimeSpan KeyVaultSecretCachingTimeout = TimeSpan.FromDays(1);
+
+        private IServiceProvider _serviceProvider;
+        private ReadCursor<DateTimeOffset> _cursor;
+        private IGitHubVulnerabilitiesLagTelemetryService _telemetryService;
+        private IGitHubQueryService _queryService;
+
+        public override void Init(IServiceContainer serviceContainer, IDictionary<string, string> jobArgsDictionary)
+        {
+            var configurationFilename = JobConfigurationManager.GetArgument(jobArgsDictionary, ConfigurationArgument);
+
+            _serviceProvider = GetServiceProvider(GetConfigurationRoot(configurationFilename));
+            _cursor = _serviceProvider.GetRequiredService<ReadCursor<DateTimeOffset>>();
+            _telemetryService = _serviceProvider.GetRequiredService<IGitHubVulnerabilitiesLagTelemetryService>();
+            _queryService = _serviceProvider.GetRequiredService<IGitHubQueryService>();
+        }
+
+        public async override Task Run()
+        {
+            var token = new CancellationToken();
+
+            try
+            {
+                var currentTimeUtc = DateTimeOffset.UtcNow;
+
+                // Query for the current cursor of GitHubVulnerabilities2Db job.
+                // We'll use this cursor to check for newer results in GitHub's API.
+                await _cursor.Load(token);
+
+                // Query GitHub for the latest security advisory UpdatedAt value beyond the current job's cursor position.
+                var latestAdvisoryUpdatedAt = await _queryService.GetLatestAdvisoryUpdatedAtValueAfterCursor(_cursor.Value, token);
+
+                // Track metric
+                _telemetryService.TrackGitHubVulnerabilities2DbLag(currentTimeUtc, _cursor.Value, latestAdvisoryUpdatedAt);
+
+            }
+            catch (Exception e)
+            {
+                Logger.LogError("Exception Occured. {Exception}", e);
```

Comment:
This may be hard to understand from the log message alone. Maybe expand the message a little?

```suggestion
                Logger.LogError("Could not track GitHub vulnerabilities lag due to exception. {Exception}", e);
```

Also, should we let the exception bubble up?